### DISCUSSION TITLE
Column casted as enum

### DISCUSSION THREAD

**Discussion by eafarooqi** [Power-Components/livewire-powergrid#1353] [General]
### Have you searched through other issues to see if your problem is already reported or has been fixed?

Yes, I did not find it.

### Did you read the documentation?

Yes, I did not find it.

### Have you tried to publish the views?

Yes - I didn't work.

### Is there an error in the console?

No

### PHP Version

8.1.0

### PowerGrid

5.3.1

### Laravel

10.42.0

### Livewire

3.4.1

### Alpine JS

3.12.0

### Theme

Bootstrap

### Describe the bug.

I have an enum field which is include in the $cast attribute as enum. As per document i am getting the following error. 
"Male" is not a valid backing value for enum "App\Enums\Person\Gender"

- cast property in model
```
protected $casts = [
        'gender'            => Gender::class,
];
```

- Gender enum
```
enum Gender: string
{
    use EnumToCollectionTrait;

    case MALE = 'M';
    case FEMALE = 'F';

    public function label(): string
    {
        return match($this) {
            self::MALE => 'Male',
            self::FEMALE => 'Female',
        };
    }
}
```

- Column 
```
->addColumn('gender', function (Person $person) {
                return Gender::tryFrom($person->gender->value)->label();
            });
```

This only works when cast is removed from the model and column is changed to as follows (as in documentation). But i cannot remove the cast.
```
return Gender::tryFrom($person->gender)->label();
```

- Have also tried with cast property as enum.
```
->addColumn('gender', function (Person $person) {
                return $person->gender->value;
            });
```
getting the following error in this case. 
preg_replace(): Argument # 3 ($subject) must be of type array|string, App\Enums\Person\Gender given

or as follows
```
 ->addColumn('gender', function (Person $person) {
                return $person->gender->label();
            });
```
The error is same as the first one. 
"Male" is not a valid backing value for enum "App\Enums\Person\Gender"

### To Reproduce...

Add enum column to the cast propery in model.

**Comment by dansysanalyst** [comment]
Hi @eafarooqi,

Following your example, add the column with a different name than `gender`, for instance `gender label`.



```php
    public function addColumns(): PowerGridColumns
    {
        return PowerGrid::columns()
            ->addColumn('id')
            ->addColumn('name')
            ->addColumn('gender_label', function (User $person) {
                return $person->gender->label();
            });
```

Then, when including the column, use the `gender_label` source:

```php
public function columns(): array
    {
        return [
            Column::make('Id', 'id'),

            Column::make('gender', 'gender_label'),
            
            //....
        ];
    }

```

You should be getting the label of your Enum in the table:

![CleanShot 2024-01-25 at 20 08 16@2x](https://github.com/Power-Components/livewire-powergrid/assets/79267265/5dd6023c-31e1-44ac-a5d0-40bdf841a921)

**Comment by eafarooqi** [comment]
Hello @dansysanalyst 

thank you so much for the solution. That i have already figured it out that the enum columns can be used with this work around. But is it a bug? or is it a limitation in this package that the enum columns cant be used in the default way ?

**Comment by elsoftpy** [comment]
Hi, I can't make the ```searchable()``` method to work though.
I've already tried adding the field with a different name from the database and making the column using the "alias" as source.
Following the above example let's say I used   ```gender_label``` and as @dansysanalyst said the table shows correctly, but adding the ```searchable()``` method to the column is not working
